feat: Checkbox, Radio 컴포넌트 구현(#15)

jngmnj opened the pull request:
## 관련 이슈
- 관련이슈: #15 

## 작업 개요
- [x] ✨feature

## 작업 상세 내용
`src/components/common/Checkbox.tsx`
`src/components/common/Radio.tsx`

- Remix Icon 기반 체크 UI 적용
- children을 통한 라벨 텍스트 커스터마이징 가능
- 접근성을 위해 input은 sr-only 처리하고 label로 전체 래핑
- peer-checked를 활용해 활성화된 상태에서 스타일 변경

### 사용법
```jsx
      <Checkbox id="checkbox-id" name="checkbox">
        체크박스
      </Checkbox>

      <div className="mt-4 flex gap-4">
        <Radio name="radio">라디오 버튼 1</Radio>
        <Radio name="radio">라디오 버튼 2</Radio>
        <Radio name="radio">라디오 버튼 3</Radio>
        <Radio name="radio">라디오 버튼 4</Radio>
      </div>
```

## 스크린샷
<img width="605" height="172" alt="스크린샷 2025-07-29 오후 6 08 24" src="https://github.com/user-attachments/assets/15a43f13-32e8-4a7b-bf81-2b81ab7eaa92" />


## 참고사항
-
-

jngmnj committed:
feat: 커스텀 체크박스 컴포넌트 구현 (#15)

- Remix Icon 기반 체크 UI 적용
- children을 통한 라벨 텍스트 커스터마이징 가능
- 접근성을 위해 input은 sr-only 처리하고 label로 전체 래핑

jngmnj committed:
feat: 접근성 고려한 커스텀 라디오 버튼 컴포넌트 추가 (#15)

- input[type="radio"]를 sr-only 처리하여 시맨틱 구조 유지
- peer-checked를 활용해 활성화된 상태에서 스타일 변경
- 원형 테두리 및 border-6로 선택 상태 강조
- RemixIcon 적용하여 아이콘 기반 인터페이스 구성
- children으로 라벨 텍스트 유연하게 지정 가능

Aprasaks commented on src/components/common/Checkbox.tsx:
```diff
@@ -0,0 +1,35 @@
+    <label className={`inline-flex cursor-pointer items-center gap-2`}>
+      <input
+        type="checkbox"
+        className="peer sr-only"
```
와 이거 peer 하고 sr-only 는 따로였네요.
peer가 상태변화되면 다른요소가 감지하는거고
sr-only가 화면에서 숨겨지는 건가요?

jngmnj replied:
넵 peer로 형제요소의 변화를 다른 요소에 스타일을 적용시켜주기위해 씁니다

idubusomuch commented on src/components/common/Checkbox.tsx:
```diff
@@ -0,0 +1,35 @@
+
+const Checkbox = ({
+  onToggle,
+  className,
```
지금 props로 받고있는 className이 사용이 안되고 있는 것 같습니다
(+혹시 확장성때문에 그냥 적어만 두신걸까요?)

jngmnj replied:
라벨텍스트의 스타일을 위해서 받으려고했는데 그냥 children으로 받기로했습니다. 
사용하지 않게되었는데 삭제가 덜돼서 이부분 수정해서 다시 올렸습니닷

jngmnj committed:
refactor: Radio 및 Checkbox 컴포넌트 스타일 정리 및 불필요한 props 제거 (#15)

- className props 제거로 불필요한 유연성 제거
- JSX 내 클래스 문자열을 간결하게 리팩토링
- 가독성 향상을 위해 불필요한 줄바꿈 및 중복 제거
